focus-android/app/src/main/java/org/mozilla/focus/autocomplete/AutocompleteListFragment.kt

```diff
@@ -308,7 +256,7 @@ open class AutocompleteListFragment : BaseSettingsLikeFragment(), CoroutineScope
             handleView.isVisible = isSelectionMode
             handleView.setOnTouchListener { _, event ->
                 if (event.actionMasked == MotionEvent.ACTION_DOWN) {
-                    itemTouchHelper.startDrag(this)
+                    itemTouchHelper?.startDrag(this)
```

Comment:
This seems to be the proposed fix for the crash?
In which case we could avoid the refactoring by just checking `isSelectionMode` before setting the `onTouchListener` or before trying to drag the item, similar to [what we do for the click listener](https://searchfox.org/mozilla-mobile/rev/9649a03c3b0044af635ad9d06eb2d3b98866ff1d/firefox-android/focus-android/app/src/main/java/org/mozilla/focus/autocomplete/AutocompleteListFragment.kt#316-320).

This got me to think about why we don't have the same behavior for the two listeners and seems like we leveraged the `handleView` not  being visible while in selection mode and as such users could not interact with it and so the `OnTouchListener` would not be called.
This behavior of setting the proper visibility for the `handleView` seems to be a regression after https://github.com/mozilla-mobile/firefox-android/commit/08a4accd95245aefee80c037142d8e73ca59f993#diff-50d6207d57d99bc04260668b6c87256cb402dafb8a2c7e67d701fe8b3adb1cd5R308. So if we set the proper visibility back then we'd solve both a UI issue and the crash.

Reply:
This was actually a fix for the effect of what had caused the crash. After a longer investigation, I discovered the cause of the problem: the itemTouchHelper was being attached to the recycler view only when selectionMode was false. The handlerView becomes visible only when selectionMode becomes true, so the itemTouchHelper wasn't configured when displaying the list of removable elements (hence the crash when trying to drag the element of the list). 
I have also tested the code before https://github.com/mozilla-mobile/firefox-android/commit/08a4accd95245aefee80c037142d8e73ca59f993#diff-50d6207d57d99bc04260668b6c87256cb402dafb8a2c7e67d701fe8b3adb1cd5R308 and the problem was still there.